cs/src/Management/IdGeneration.cs

```diff
@@ -0,0 +1,26 @@
+    private static readonly Random random = new Random();
+    /// <summary>
+    /// Generate valid tunnelIds
+    /// </summary>
+    /// <returns>string tunnel id</returns>
+    public static string GenerateTunnelId()
+    {
+        var sb = new StringBuilder();
+        sb.Append(adjectives[random.Next(adjectives.Length)]);
+        sb.Append("-");
+        sb.Append(nouns[random.Next(nouns.Length)]);
+        return sb.ToString();
```

Comment:
Should we check if the result is not exceeding the max id length?

Reply:
Based on the small set of words we cannot exceed the length